### PR TITLE
Refactor System Form

### PR DESCRIPTION
**Pull request by church29**
Solves https://github.com/OSU-Net/cyder/issues/783, https://github.com/OSU-Net/cyder/issues/784, https://github.com/OSU-Net/cyder/issues/671, https://github.com/OSU-Net/cyder/issues/809

### CODE REVIEW

**Comment by drkitty** [comment]
On master, creations forms have an empty `action` attribute. (Apparently the attribute is only set on update forms for some reason.) However, on this branch, creations forms have no `action` attribute at all, which causes an error on [this line](https://github.com/OSU-Net/cyder/blob/8fee1347e6/media/js/views.js#L198) because `undefined` has no `indexOf` method.

**Comment by drkitty** [comment]
Looks like this doesn't yet resolve issues #783 and #784, because the system creation form still has its own page if you click "Create System" on the system list page.

**Comment by drkitty** [comment]
On the system creation form, the POST request sent by the submit button is sent to `/core/system/`, which is the wrong URL. I think this has to do with the missing `action` attribute.

**Comment by church29** [comment]
@drkitty I fixed the issue you mentioned, but ran into another. Try creating a interface from a range detail view.

**Comment by church29** [comment]
Hmm, it seems to be working fine now. I think my local instance at home needs to be rebuilt.

**Comment by drkitty** [comment]
Everything appears to work properly.

File: media/js/views.js [hunk 1]
```diff
@@ -41,13 +41,11 @@ $(document).ready( function() {
...
     // handles create buttons in dynamic/static interface view and range
     // detail view
-    $( document ).on( 'click', '#system_create', function( e ) {
-        e.preventDefault();
-        var csrfToken = $('#view-metadata').attr( 'data-csrfToken' );
-        button_to_form( this, csrfToken, function( postForm ){
-            $(postForm).submit();
-        });
-    });
+    //$( document ).on( 'click', '#system_create', function( e ) {
```
Comment (drkitty) [inline media/js/views.js]: Why is this commented out? If it's not doing anything I think it should just be removed.

File: media/js/views.js [hunk 2]
```diff
@@ -80,15 +78,19 @@ $(document).ready( function() {
...
         } else {
-            $('.js-get-form, .js-create-object, .update, .cancel').removeClass( 'selected' );
+            $(selectors).removeClass( 'selected' );
             $(this).removeClass( 'hover' ).addClass( 'selected' );
         }
     }
-    $( document ).on('click', '.js-get-form, js-create-object, .update, .cancel', buttonLogic );
+    $( document ).on('click',
+        '.js-get-form, .js-create-object, .update, .cancel, .system_form',
+        buttonLogic );
```
Comment (drkitty) [inline media/js/views.js]: Any reason not to inline this function?

Comment (church29) [inline media/js/views.js]: I am gonna leave as is for now. I am gonna do a simple pr to add a js-buttonLogic selector.

**Comment by drkitty** [comment]
I'll do a final review in an hour or two and then this should be ready for merge.

**Comment by drkitty** [comment]
If you go to the dynamic interface list page and click "Create dynamic interface", the form that pops up shows both static and dynamic ranges. It's supposed to show only dynamic ranges. It works correctly from the system create page.

**Comment by church29** [comment]
Ah, good catch. Should be a fairly easy fix.

**Comment by church29** [comment]
Fixed.

**Comment by drkitty** [comment]
That works for the initial range list shown, but don't forget that the range wizard replaces the range choices, and that still doesn't filter properly even with the latest commit.

**Comment by church29** [comment]
Did a quick fix for that. I think I might have broke rangeWizard for non interfaces in my js refactor pr. So lets hold off on merging this. I wanna clean things up.

**Comment by drkitty** [comment]
Regression: The "name" field on the combined system creation form and every text field on the static and dynamic interface creation and edit forms has lost its `keypress` event handler that's supposed to submit the form when the enter key is pressed.

**Comment by drkitty** [comment]
The text fields other than "name" on the combined interface creation form have been that way for a while, so that's just a bug, not a regression.

**Comment by drkitty** [comment]
Everything else seems to function properly. Nice work on this branch.

**Comment by church29** [comment]
The keypress event handler is never attached with initForms(). The last inputs on the dynamic/static forms are either going to be a dropdown or a checkbox so it doesn't really apply.

**Comment by akeym** [comment]
Is this ready for review, or is it just ready to ship?

**Comment by church29** [comment]
This is ready to ship.

**Comment by drkitty** [comment]
So should I create a separate issue for the fields that don't cause form submission when you press Enter?

**Comment by church29** [comment]
Yes, it would be a lot easier to do all of the submission issues at once.

**Comment by drkitty** [comment]
Alright. I'll go ahead and merge this, then.

**Pull request merged**